blog/views.py

```diff
@@ -52,81 +53,116 @@ def update(self, request, *args, **kwargs):
 
 
 
-class CategoryCreate(generics.ListCreateAPIView):
+# class CategoryCreate(generics.ListCreateAPIView):
 
-    """ create a category """
+#     """ create a category """
 
-    queryset = Category.objects.all()
-    serializer_class = CategorySerializers
+#     queryset = Category.objects.all()
+#     serializer_class = CategorySerializers
 
-class CategoryDetail(generics.RetrieveUpdateDestroyAPIView):
-    """ list all categories """
+# class CategoryDetail(generics.RetrieveUpdateDestroyAPIView):
+#     """ list all categories """
 
+#     queryset = Category.objects.all()
+#     serializer_class = CategorySerializers
+
+class CategoryViewset(viewsets.ModelViewSet):
     queryset = Category.objects.all()
     serializer_class = CategorySerializers
 
+    
+    def get_permissions(self):
+        # admin is allowed to destroy and update a category
+        if self.action == 'destroy' or self.action == "update":
+            self.permission_classes = [IsAdminUser]
+        # anyone can create, retrieve and list all categories
+        elif self.action == 'retrieve' or self.action == "create" or self.action == "list":
+            self.permission_classes = [AllowAny]
```

Comment:
We'll need to include permisions that only allow account owners to list their accounts
- [ ] Restrict only acount owners and admins to list accounts